### ISSUE TITLE
Error when using Plotly backend and adding edge labels

### ISSUE THREAD

**Issue by szhorvat** [igraph/python-igraph#689]
**Describe the bug**

Using the Plotly backend and edge labels results in an error:

```
AttributeError: 'igraph.Edge' object has no attribute 'curved'
```

**To reproduce**

```python
ig.plot(ig.Graph([(0,1), (1,2)]), edge_label=['a', 'b'], backend="plotly")
```

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[2], line 1
----> 1 ig.plot(ig.Graph([(0,1), (1,2)]), edge_label=['a','b'],  backend="plotly")

File ~/opt/miniconda3/envs/igold/lib/python3.9/site-packages/igraph/drawing/__init__.py:267, in plot(obj, target, bbox, *args, **kwds)
    265     return
    266 else:
--> 267     result = plotter(
    268         backend,
    269         target,
    270         palette=palette,
    271         *args,
    272         **kwds,
    273     )
    274     # NOTE: for matplotlib, result is the container Artist. It would be
    275     # good to return this instead of target, like we do for Cairo.
    276     # However, that breaks API so let's wait for a major release
    278     if save_path is not None:

File ~/opt/miniconda3/envs/igold/lib/python3.9/site-packages/igraph/drawing/graph.py:558, in __plot__(self, backend, context, *args, **kwds)
    552 from igraph.drawing import DrawerDirectory
    554 drawer = kwds.pop(
    555     "drawer_factory",
    556     DrawerDirectory.resolve(self, backend)(context),
    557 )
--> 558 return drawer.draw(self, *args, **kwds)

File ~/opt/miniconda3/envs/igold/lib/python3.9/site-packages/igraph/drawing/plotly/graph.py:260, in PlotlyGraphDrawer.draw(self, graph, *args, **kwds)
    258 src, dest = edge.tuple
    259 src_vertex, dest_vertex = vertex_builder[src], vertex_builder[dest]
--> 260 (x, y), (halign, valign) = edge_drawer.get_label_position(
    261     edge,
    262     src_vertex,
    263     dest_vertex,
    264 )
    265 if label is None:
    266     continue

File ~/opt/miniconda3/envs/igold/lib/python3.9/site-packages/igraph/drawing/baseclasses.py:154, in AbstractEdgeDrawer.get_label_position(self, edge, src_vertex, dest_vertex)
    151     angle = None
    153 # Determine the midpoint
--> 154 if edge.curved:
    155     (x1, y1), (x2, y2) = src_vertex.position, dest_vertex.position
    156     aux1, aux2 = get_bezier_control_points_for_curved_edge(
    157             x1, y1, x2, y2, edge.curved,
    158         )

AttributeError: 'igraph.Edge' object has no attribute 'curved'
```

**Version information**

Current `main`

<details>
<summary> Versions</summary>

```
$ conda list                                                                                         (igold) 
# packages in environment at /Users/szhorvat/opt/miniconda3/envs/igold:
#
# Name                    Version                   Build  Channel
aiofiles                  22.1.0           py39hecd8cb5_0  
aiosqlite                 0.18.0           py39hecd8cb5_0  
anyio                     3.5.0            py39hecd8cb5_0  
appnope                   0.1.2           py39hecd8cb5_1001  
argon2-cffi               21.3.0             pyhd3eb1b0_0  
argon2-cffi-bindings      21.2.0           py39hca72f7f_0  
asttokens                 2.0.5              pyhd3eb1b0_0  
attrs                     22.1.0           py39hecd8cb5_0  
babel                     2.11.0           py39hecd8cb5_0  
backcall                  0.2.0              pyhd3eb1b0_0  
beautifulsoup4            4.12.2           py39hecd8cb5_0  
blas                      1.0                         mkl  
bleach                    4.1.0              pyhd3eb1b0_0  
brotli                    1.0.9                hca72f7f_7  
brotli-bin                1.0.9                hca72f7f_7  
brotlipy                  0.7.0           py39h9ed2024_1003  
ca-certificates           2023.05.30           hecd8cb5_0  
certifi                   2023.5.7         py39hecd8cb5_0  
cffi                      1.15.1           py39h6c40b1e_3  
charset-normalizer        2.0.4              pyhd3eb1b0_0  
comm                      0.1.2            py39hecd8cb5_0  
contourpy                 1.0.5            py39haf03e11_0  
cryptography              39.0.1           py39hf6deb26_2  
cycler                    0.11.0             pyhd3eb1b0_0  
debugpy                   1.5.1            py39he9d5cce_0  
decorator                 5.1.1              pyhd3eb1b0_0  
defusedxml                0.7.1              pyhd3eb1b0_0  
entrypoints               0.4              py39hecd8cb5_0  
executing                 0.8.3              pyhd3eb1b0_0  
fonttools                 4.25.0             pyhd3eb1b0_0  
freetype                  2.12.1               hd8bbffd_0  
giflib                    5.2.1                h6c40b1e_3  
icu                       58.2                 h0a44026_3  
idna                      3.4              py39hecd8cb5_0  
igraph                    0.10.5                   pypi_0    pypi
importlib-metadata        6.0.0            py39hecd8cb5_0  
importlib_metadata        6.0.0                hd3eb1b0_0  
importlib_resources       5.2.0              pyhd3eb1b0_1  
intel-openmp              2023.1.0         ha357a0b_43547  
ipykernel                 6.19.2           py39h01d92e1_0  
ipython                   8.12.0           py39hecd8cb5_0  
ipython_genutils          0.2.0              pyhd3eb1b0_1  
jedi                      0.18.1           py39hecd8cb5_1  
jinja2                    3.1.2            py39hecd8cb5_0  
jpeg                      9e                   h6c40b1e_1  
json5                     0.9.6              pyhd3eb1b0_0  
jsonschema                4.17.3           py39hecd8cb5_0  
jupyter_client            8.1.0            py39hecd8cb5_0  
jupyter_core              5.3.0            py39hecd8cb5_0  
jupyter_events            0.6.3            py39hecd8cb5_0  
jupyter_server            2.5.0            py39hecd8cb5_0  
jupyter_server_fileid     0.9.0            py39hecd8cb5_0  
jupyter_server_terminals  0.4.4            py39hecd8cb5_1  
jupyter_server_ydoc       0.8.0            py39hecd8cb5_1  
jupyter_ydoc              0.2.4            py39hecd8cb5_0  
jupyterlab                3.6.3            py39hecd8cb5_0  
jupyterlab_pygments       0.1.2                      py_0  
jupyterlab_server         2.22.0           py39hecd8cb5_0  
kiwisolver                1.4.4            py39hcec6c5f_0  
lcms2                     2.12                 hf1fd2bf_0  
lerc                      3.0                  he9d5cce_0  
libbrotlicommon           1.0.9                hca72f7f_7  
libbrotlidec              1.0.9                hca72f7f_7  
libbrotlienc              1.0.9                hca72f7f_7  
libcxx                    14.0.6               h9765a3e_0  
libdeflate                1.17                 hb664fd8_0  
libffi                    3.4.4                hecd8cb5_0  
libiconv                  1.16                 hca72f7f_2  
libpng                    1.6.39               h6c40b1e_0  
libsodium                 1.0.18               h1de35cc_0  
libtiff                   4.5.0                hcec6c5f_2  
libwebp                   1.2.4                hf6ce154_1  
libwebp-base              1.2.4                h6c40b1e_1  
libxml2                   2.10.3               h930c0e2_0  
libxslt                   1.1.37               h6d1eb0e_0  
lxml                      4.9.2            py39h6c40b1e_0  
lz4-c                     1.9.4                hcec6c5f_0  
markupsafe                2.1.1            py39hca72f7f_0  
matplotlib                3.7.1            py39hecd8cb5_1  
matplotlib-base           3.7.1            py39hda11e5a_1  
matplotlib-inline         0.1.6            py39hecd8cb5_0  
mistune                   0.8.4           py39h9ed2024_1000  
mkl                       2023.1.0         h59209a4_43558  
mkl-service               2.4.0            py39h6c40b1e_1  
mkl_fft                   1.3.6            py39h07fba90_1  
mkl_random                1.2.2            py39h07fba90_1  
munkres                   1.1.4                      py_0  
nbclassic                 0.5.5            py39hecd8cb5_0  
nbclient                  0.5.13           py39hecd8cb5_0  
nbconvert                 6.5.4            py39hecd8cb5_0  
nbformat                  5.7.0            py39hecd8cb5_0  
ncurses                   6.4                  hcec6c5f_0  
nest-asyncio              1.5.6            py39hecd8cb5_0  
notebook                  6.5.4            py39hecd8cb5_0  
notebook-shim             0.2.2            py39hecd8cb5_0  
numpy                     1.25.0           py39h827a554_0  
numpy-base                1.25.0           py39ha186be2_0  
openssl                   3.0.9                hca72f7f_0  
packaging                 23.0             py39hecd8cb5_0  
pandocfilters             1.5.0              pyhd3eb1b0_0  
parso                     0.8.3              pyhd3eb1b0_0  
pexpect                   4.8.0              pyhd3eb1b0_3  
pickleshare               0.7.5           pyhd3eb1b0_1003  
pillow                    9.4.0            py39hcec6c5f_0  
pip                       23.1.2           py39hecd8cb5_0  
platformdirs              2.5.2            py39hecd8cb5_0  
plotly                    5.9.0            py39hecd8cb5_0  
prometheus_client         0.14.1           py39hecd8cb5_0  
prompt-toolkit            3.0.36           py39hecd8cb5_0  
psutil                    5.9.0            py39hca72f7f_0  
ptyprocess                0.7.0              pyhd3eb1b0_2  
pure_eval                 0.2.2              pyhd3eb1b0_0  
pycparser                 2.21               pyhd3eb1b0_0  
pygments                  2.15.1           py39hecd8cb5_1  
pyopenssl                 23.0.0           py39hecd8cb5_0  
pyparsing                 3.0.9            py39hecd8cb5_0  
pyrsistent                0.18.0           py39hca72f7f_0  
pysocks                   1.7.1            py39hecd8cb5_0  
python                    3.9.16               h5ee71fb_3  
python-dateutil           2.8.2              pyhd3eb1b0_0  
python-fastjsonschema     2.16.2           py39hecd8cb5_0  
python-json-logger        2.0.7            py39hecd8cb5_0  
pytz                      2022.7           py39hecd8cb5_0  
pyyaml                    6.0              py39h6c40b1e_1  
pyzmq                     25.1.0           py39hcec6c5f_0  
readline                  8.2                  hca72f7f_0  
requests                  2.29.0           py39hecd8cb5_0  
rfc3339-validator         0.1.4            py39hecd8cb5_0  
rfc3986-validator         0.1.1            py39hecd8cb5_0  
send2trash                1.8.0              pyhd3eb1b0_1  
setuptools                67.8.0           py39hecd8cb5_0  
six                       1.16.0             pyhd3eb1b0_1  
sniffio                   1.2.0            py39hecd8cb5_1  
soupsieve                 2.4              py39hecd8cb5_0  
sqlite                    3.41.2               h6c40b1e_0  
stack_data                0.2.0              pyhd3eb1b0_0  
tbb                       2021.8.0             ha357a0b_0  
tenacity                  8.2.2            py39hecd8cb5_0  
terminado                 0.17.1           py39hecd8cb5_0  
texttable                 1.6.7                    pypi_0    pypi
tinycss2                  1.2.1            py39hecd8cb5_0  
tk                        8.6.12               h5d9f67b_0  
tomli                     2.0.1            py39hecd8cb5_0  
tornado                   6.2              py39hca72f7f_0  
traitlets                 5.7.1            py39hecd8cb5_0  
typing-extensions         4.6.3            py39hecd8cb5_0  
typing_extensions         4.6.3            py39hecd8cb5_0  
tzdata                    2023c                h04d1e81_0  
urllib3                   1.26.16          py39hecd8cb5_0  
wcwidth                   0.2.5              pyhd3eb1b0_0  
webencodings              0.5.1            py39hecd8cb5_1  
websocket-client          0.58.0           py39hecd8cb5_4  
wheel                     0.38.4           py39hecd8cb5_0  
xz                        5.4.2                h6c40b1e_0  
y-py                      0.5.9            py39h7242b5c_0  
yaml                      0.2.5                haf1e3a3_0  
ypy-websocket             0.8.2            py39hecd8cb5_0  
zeromq                    4.3.4                h23ab428_0  
zipp                      3.11.0           py39hecd8cb5_0  
zlib                      1.2.13               h4dc903c_0  
zstd                      1.5.5                hc035e20_0  
```
</details>

**Comment by ntamas** [comment]
A temporary fix is now in place in 35a4ba43. Label colors do not work yet and I'm not familiar with `plotly` enough to figure it out in a short timeframe, but at least it's not crashing now.

**Comment by szhorvat** [comment]
Open a new issue for the missing piece so it wouldn't be forgotten.
